ballerina/field.bal

```diff
@@ -160,4 +178,33 @@ public class Field {
     isolated function getFieldInterceptors() returns readonly & Interceptor[] {
         return self.fieldInterceptors;
     }
+
+    isolated function isCacheEnabled() returns boolean {
+        return self.cacheEnabled;
+    }
+
+    isolated function getCacheConfig() returns ServerCacheConfig? {
+        return self.cacheConfig;
+    }
+
+    isolated function getCacheKey() returns string {
+        return self.generateCacheKey();
+    }
+
+    isolated function getCacheMaxAge() returns decimal {
+        return self.cacheMaxAge;
+    }
+
+    isolated function getParentArgHashes() returns readonly & string[] {
+        return self.parentArgHashes;
+    }
+
+    private isolated function generateCacheKey() returns string {
+        string resourcePath = "";
+        foreach string|int path in self.path {
+            resourcePath += string `${path}.`;
+        }
+        string hash = generateArgHash(self.internalNode.getArguments(), self.parentArgHashes);
+        return string `${resourcePath}${hash}`;
```

Comment:
Why do we need `${resourcePath}${hash}` for the cache key? Isn't just `hash` is enough?

Reply:
The resource path is used to find the cache entries when manually evicting as mentioned in the proposal.

Reply:
Even when manually evicting, we can generate the hash and check for the key, right? 
Using the `resource path` + `hash` beats the purpose of hashing. We used hash to reduce the size of the keys. But now we are creating even longer keys.

Reply:
When manually evicting, how can we generate the cache key related to a specific cache entry without knowing its arguments(if the cached resource function has arguments)?